ui/jetbrains/src/main/java/com/neueda/jetbrains/plugin/graphdb/jetbrains/ui/datasource/metadata/tree/Root.java

```diff
@@ -0,0 +1,55 @@
+package com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.metadata.tree;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class Root implements TreeNode, TreeNodeDirectory {
+
+    private final DefaultMutableTreeNode root;
+    private final Map<String, Directory> directories;
+    private final List<Leaf> functions;
+    private final Function<TreeNode, DefaultMutableTreeNode> directoryProducer;
+    private final Function<TreeNode, DefaultMutableTreeNode> functionProducer;
+
+    public Root(
+            DefaultMutableTreeNode root,
+            Function<TreeNode, DefaultMutableTreeNode> directoryProducer,
+            Function<TreeNode, DefaultMutableTreeNode> functionProducer
+    ) {
+        this.root = root;
+        this.directoryProducer = directoryProducer;
+        this.functionProducer = functionProducer;
+        this.directories = new LinkedHashMap<>();
+        this.functions = new ArrayList<>();
+    }
+
+    @Override
+    public Directory getDirectory(String name) {
+        return directories.computeIfAbsent(
+                name,
+                (k) -> new Directory(k, directoryProducer, functionProducer)
```

Comment:
I can't understand what is 'k'. Please use full names for the variables

Reply:
'k' stands for key, however I changed it to 'newName' for more understanding